clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBStorageEngineFactory.java

```diff
@@ -275,6 +295,34 @@ public synchronized void removeStorageEngine(AbstractStorageEngine engine) {
     }
   }
 
+  /**
+   * The following function shouldn't be invoked at runtime (only at startup time), so we didn't apply
+   * any throttling here, and if the above condition changes in the future, we may need to consider
+   * throttling the deletion to reduce the IO impact to the database disk, which may result in some
+   * side effect in the read path.
+   */
+  @Override
+  public synchronized void removeStorageEngine(String storeName) {
```

Comment:
As far as I can see, it is only invoked in restoreAllStores, Just wondering why we could not reuse the existing API removeStorageEngine(AbstractStoreageEngine engine)?

Reply:
Good question.
Today, sst file manager doesn't check space limit when opening a new database, which means DaVinci should be able to open the database even its size is bigger than memory limit.
But this is not a good long-term solution since database opening should check the space limit as well, and I am planning to work with RocksDB team to fix this issue.
Once that fix is in, opening up a storage engine and then deleting it won't work.

Reply:
Got it.